lib/elliptic/eddsa/index.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+var bn = require('bn.js');
+var hash = require('hash.js');
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+
+function parseBytes(bytes) {
+  return typeof bytes === 'string' ? utils.toArray(bytes, 'hex') :
+                                     bytes;
+}
+
+function EDDSA(curve) {
+  if (!(this instanceof EDDSA))
+    return new EDDSA(curve);
+
+  assert(typeof curve === 'string');
+  assert(curve === 'ed25519', 'only tested with ed25519 so far');
+  var curve = this.curve = elliptic.curves[curve].curve;
+  this.g = curve.g;
+  // Later we do a G.mul(fiveHundredAnd12BitNumber) so we need more doubles.
```

Comment:
By very quick crude timing of `npm test`, it does seem be a little bit faster ( ~12% ) when using `_fixedNafMul`